incubator/hnc/internal/reconcilers/object_test.go

```diff
@@ -59,6 +59,22 @@ var _ = Describe("Secret", func() {
 		Consistently(hasObject(ctx, "Role", bazName, "testrole")).Should(BeFalse())
 	})
 
+	It("should remove object in selected namespace when treeSelector is updated", func() {
+		setParent(ctx, barName, fooName)
+		setParent(ctx, bazName, fooName)
+
+		makeObject(ctx, "Role", fooName, "testrole")
+		Eventually(hasObject(ctx, "Role", barName, "testrole")).Should(BeTrue())
+		Eventually(hasObject(ctx, "Role", bazName, "testrole")).Should(BeTrue())
```

Comment:
I've just discovered (in #1174) that you can comment the "Should" clause, e.g.:

`Eventually(hasObject(ctx, "Role", barName, "testrole)).Should(BeTrue(), "When propagating testrole to %s", barName)`

Maybe start annotating all your `Should` clauses like this so the errors are more descriptive?

Reply:
done